packages/indexer/src/index.ts

```diff
@@ -4,13 +4,40 @@ import { Indexer as IndexerRpc } from './indexer.gen'
 
 const fetch = typeof global === 'object' ? global.fetch : window.fetch
 
+export interface SequenceIndexerOpts {
+  JWTAuth?: string
+  ProjectAccessKey?: string
+}
+
 export class SequenceIndexer extends IndexerRpc {
+  public jwtAuth?: string
+  public projectAccessKey?: string
+
   constructor(
     hostname: string,
-    public projectAccessKey?: string,
-    public jwtAuth?: string
+    opts: SequenceIndexerOpts
+  );
+  constructor(
+    hostname: string,
+    projectAccessKey?: string,
+    jwtAuth?: string
+  );
+  constructor(
+    hostname: string,
+    ...args: any[]
   ) {
     super(hostname.endsWith('/') ? hostname.slice(0, -1) : hostname, fetch)
+
+    if (args.length === 1 && typeof args[0] === 'object') {
+      const opts = args[0] as SequenceIndexerOpts
+      this.projectAccessKey = opts.ProjectAccessKey
+      this.jwtAuth = opts.JWTAuth
+    } else {
+      const [projectAccessKey, jwtAuth] = args
+      this.projectAccessKey = projectAccessKey || undefined
+      this.jwtAuth = jwtAuth || undefined
```

Comment:
this is for backward-compatibility?

can we `console.error()` with a warning to update to the new Opts object?

was this implemented a long time ago? do you expect customers using it?